operator/v1/types_ingress.go

```diff
@@ -483,6 +483,46 @@ type LoggingDestination struct {
 	Container *ContainerLoggingDestinationParameters `json:"container,omitempty"`
 }
 
+// IngressControllerCaptureHTTPHeader describes an HTTP header that should be
+// captured.
+type IngressControllerCaptureHTTPHeader struct {
+	// name specifies a header name.  Its value must be a valid HTTP header
+	// name as defined in RFC 2616 section 4.2.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	// +required
+	Name string `json:"name"`
+
+	// maxLength specifies a maximum length for the header value.  If a
+	// header value exceeds this length, the value will be truncated in
+	// the log message.
+	//
+	// +kubebuilder:validation:Required
```

Comment:
greater than zero I presume

Reply:
I think it could be equal to zero (which would allow logging the presence of the header whilst dropping the value); from the code, it looks like HAProxy allows this, but I'll check in a running cluster today.